### PR TITLE
Validate table exists in JdbcSqlConnector [HZ-2286]

### PR DESCRIPTION
**Pull request by frant-hartm**
When CREATE MAPPING has no fields defined and a table does not exist the mapping was created with no fields.

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java [hunk 1]
```diff
@@ -150,6 +151,25 @@ private Map<String, DbField> readDbFields(
         }
     }
 
+    private static void checkTableExists(ExternalJdbcTableName externalTableName, DatabaseMetaData databaseMetaData)
+            throws SQLException {
+
+        ResultSet tables = databaseMetaData.getTables(
```
Comment (orcunc) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java:157]: The resulSet needs to be closed.
````
try(ResultSet rs =databaseMetaData.getTables(...})) {
  if (rs.next()) {
   ...
  } else {
  throw new HazelcastException(...);
  }
}
````

**Review by orcunc — CHANGES_REQUESTED** [review]

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/MappingJdbcSqlConnectorTest.java [hunk 2]
```diff
@@ -162,6 +162,21 @@ public void createMappingTableDoesNotExist() {
         );
     }
 
+    @Test
+    public void createMappingNoColumnsTableDoesNotExist() {
+        tableName = randomTableName();
+        assertThatThrownBy(() -> execute(
+                "CREATE MAPPING " + tableName + " "
+                        + "DATA CONNECTION " + TEST_DATABASE_REF
```
Comment (TomaszGaweda) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/MappingJdbcSqlConnectorTest.java:170]:
```suggestion
                "CREATE MAPPING " + tableName + " DATA CONNECTION " + TEST_DATABASE_REF
```

File: hazelcast-sql/src/main/java/com/hazelcast/sql/impl/QueryUtils.java [hunk 3]
```diff
@@ -70,15 +72,17 @@ public static String wrapDataConnectionKey(String dataConnectionKey) {
...
     }
 
     /**
      * Create map from member ID to owned partitions.
      *
      * @param nodeEngine                node engine
-     * @param localMemberVersion        version of the local member. If any of partition owners have a different version,
+     * @param localMemberVersion        version of the local member. If any of partition owners have a different
+     *                                  version,
      *                                  an exception is thrown. The check is ignored if passed version is {@code null}
```
Comment (TomaszGaweda) [inline hazelcast-sql/src/main/java/com/hazelcast/sql/impl/QueryUtils.java:86]:
```suggestion
     *                                  version, an exception is thrown. 
     *                                  The check is ignored if passed version is {@code null}
```

**Review by TomaszGaweda — APPROVED** [review]
super small nits only

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java (hunk 1 above)
Comment (frant-hartm) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java]: Good catch.
2ff83c6b8c3

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/MappingJdbcSqlConnectorTest.java (hunk 2 above)
Comment (frant-hartm) [reply hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/MappingJdbcSqlConnectorTest.java]: 2ff83c6b8c3

**Review by orcunc — APPROVED** [review]

**Pull request merged**